### PR TITLE
add volumebindingmode

### PR DESCRIPTION
**Pull request by ropatil010**
Hi Team,

Kindly PTAL.

Discussion: https://redhat-internal.slack.com/archives/C02221SB07R/p1686315404107699

Logs:
AWS: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/811536/console
GCP: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/811537/console

/assign @chao007 @duanwei33 @Phaow @radeore

### CODE REVIEW

**Comment by ropatil010** [comment]
Sure thanks @duanwei33, I have added only for vsphere platform with immediate volumebinding mode.
But the problem is when executing tc: 22882, the code again replaces with Immediate which should not be the case, even though in code there is line unless path.  Any way to check if it does not have the parameter volumeBindingMode in tc level and then add in *.rb file? 
 table.raw.each do |path, value|
      eval "sc_hash#{path} = YAML.load value" unless path == ''
  end

File: features/step_definitions/storage_class.rb [hunk 1]
```diff
@@ -261,7 +261,15 @@
       eval "sc_hash#{path} = YAML.load value" unless path == ''
   end
 
-  # replace the provisioner value according to platform wise 
+  if !sc_hash.dig("volumeBindingMode")
```
Comment (duanwei33) [inline features/step_definitions/storage_class.rb:264]: Could we add one comment/expected result for this new adding?

**Comment by duanwei33** [comment]
lgtm
Thanks for the update.

File: features/step_definitions/storage_class.rb [hunk 2]
```diff
@@ -261,6 +261,15 @@
       eval "sc_hash#{path} = YAML.load value" unless path == ''
   end
 
+  # if no volumeBindingMode exists in tc, we need to pass vSphere=Immediate, others=WaitForFirstConsumer
+  if !sc_hash.dig("volumeBindingMode")
+    if platform == "vsphere"  
+      volumeBindingMode = "Immediate"
```
Comment (liangxia) [inline features/step_definitions/storage_class.rb:267]: I think it should be `sc_hash["volumeBindingMode"] = "Immediate"` here.

**Comment by ropatil010** [comment]
Fixed. 
Logs: 
  AWS: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp- 
  common/job/Runner/812317/console
  GCP: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp- 
  common/job/Runner/812319/console
  VS: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp- 
  common/job/Runner/812318/console

/assign @liangxia @duanwei33

**Pull request merged**